teku/src/main/java/tech/pegasys/teku/AbstractNode.java

```diff
@@ -154,7 +154,12 @@ private void pollRejectedExecutions() {
   @Override
   public void stop() {
     // Stop processing new events
-    eventChannels.stop();
+    eventChannels
+        .stop()
+        .orTimeout(30, TimeUnit.SECONDS)
+        .handleException(error -> LOG.error("Failed to stop event channels", error))
```

Comment:
not sure that we want an error here in all cases... its likely mostly noise?

Reply:
hmm, not sure. Given this is how during normal shutdown we'll wind up ensuring that any async updates actually get written to the DB I think we want to know if that fails. This wouldn't log an error if any of the tasks fail, just if the event channels can't finish processing events within the 30 seconds.  I might bump it down to WARN since I think we want to know about it but it's probably not the end of the world.